test/analytics.test.ts

```diff
@@ -0,0 +1,147 @@
+  }
+
+  beforeEach(() => {
+    // Reset mocks and environment before each test
+    vi.resetAllMocks()
+    process.env = { ...originalEnv }
+    console.error = vi.fn()
+
+    // Set up environment variables
+    process.env.GITHUB_ACTION = 'test-action'
+    process.env.GITHUB_WORKFLOW = 'test-workflow'
+  })
+
+  afterEach(() => {
+    // Restore original environment and console.error after each test
+    process.env = originalEnv
+    console.error = originalConsoleError
+    vi.clearAllMocks()
+  })
+
+  describe('sendAnalytics', () => {
+    it('should send analytics data with correct properties', async () => {
+      // We need to use a dynamic import to avoid hoisting issues
+      const { sendAnalytics } = await import('../src/analytics')
+
+      await sendAnalytics(mockUserSpecifiedData)
+
+      // Check that the API request was made with the right data
+      expect(apiUtils.apiRequest).toHaveBeenCalledTimes(1)
+      expect(apiUtils.apiRequest).toHaveBeenCalledWith(
+        expect.any(String),
+        {
+          method: 'POST',
+          body: expect.stringContaining('"source":"github-action"'),
+        },
+        expect.objectContaining({
+          maxRetries: 1,
+        })
+      )
+    })
+
+    it('should use the default analytics URL if not provided in environment', async () => {
+      delete process.env.GRAFANA_ANALYTICS_URL
+
+      // We need to use a dynamic import to avoid hoisting issues
+      const { sendAnalytics } = await import('../src/analytics')
+
+      await sendAnalytics(mockUserSpecifiedData)
+
+      expect(apiUtils.apiRequest).toHaveBeenCalledWith(
+        'https://stats.grafana.org',
+        expect.any(Object),
+        expect.any(Object)
+      )
+    })
+
+    it('should use the custom analytics URL if provided in environment', async () => {
+      process.env.GRAFANA_ANALYTICS_URL = 'https://custom-stats.example.com'
+
+      // We need to use a dynamic import to avoid hoisting issues
+      const { sendAnalytics } = await import('../src/analytics')
+
+      await sendAnalytics(mockUserSpecifiedData)
+
+      expect(apiUtils.apiRequest).toHaveBeenCalledWith(
+        'https://custom-stats.example.com',
+        expect.any(Object),
+        expect.any(Object)
+      )
+    })
+
+    it('should handle errors when sending analytics', async () => {
+      // Mock apiRequest to throw an error
+      vi.mocked(apiUtils.apiRequest).mockRejectedValueOnce(
+        new Error('Network error')
+      )
+
+      // We need to use a dynamic import to avoid hoisting issues
+      const { sendAnalytics } = await import('../src/analytics')
+
+      // Call the function
+      await sendAnalytics(mockUserSpecifiedData)
+
+      // Verify that the error was handled and logged
+      expect(console.error).toHaveBeenCalledWith(
+        'Error sending analytics:',
+        expect.any(Error)
+      )
+    })
+
+    it('should generate a unique usage stats ID when GitHub variables are missing', async () => {
```

Comment:
Should the function generate a random ID if the variables are missing? Currently, it generates a hash for empty strings, which will be the same for all executions.